### DISCUSSION TITLE
uyuni terraform integration

### DISCUSSION THREAD

**Discussion by rjmateus** [uyuni-project/uyuni#3003] [Ideas]
Check this wiki page with ideas on how to integrate [uyuni with terraform](https://github.com/uyuni-project/uyuni/wiki/uyuni-terraform-integration).

Main goal is how to help ones using terraform to bring their machines to be managed on Uyuni.

**Comment by RenownTwiz** [comment]
Thanks for putting together the page on this topic.

We've been tinkering with the idea of writing a Terraform provider because we've been dealing with this issue for awhile.  We do bootstrap the VMs using Puppet during our Terraform builds so we don't need the provider to handle the bootstrap at the moment.  Where we're having issues is on the decommission/destroy of the VM/server.  There isn't really a good way to trigger a cleanup in Uyuni to remove it from the Uyuni side (The VM is gone so Puppet isn't running anymore...), the only way we see this happening is with a Terraform provider.  Do you have any thoughts on this?

**Reply by rjmateus** [reply]
This could be a good use case to make sure the machine gets removed from uyuni as soon it gets deleted. another indirect approach could be develop a process calling the API that could remove machines not responding for more then X days (not sure if it fits your use case).

**Reply by RenownTwiz** [reply]
Yes, interesting idea.  It probably would be a good Uyuni feature enhancement request.  I would think even if we don't use it others would.  Having the option to expire and delete systems that are not responding for more than X days, does make sense.  I've seen this implemented in other solutions.
